### PR TITLE
UI: Fix transform dialog not being closable

### PR DESCRIPTION
**Pull request by cg2121**
### Description
If the user clicked away from the source, while the transform dialog is open, the whole dialog would be disabled. The user then couldn't click close or press escape to close the dialog. The only way for them to close it was to click on another source to re-enable it.

### Motivation and Context
Fixes bug I've found.

### How Has This Been Tested?
Click away from a source and made sure the dialog was closable.

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by sebastian-s-beckmann — COMMENTED** [review]
I think the better fix here would be encapsulating `formLayout` in another `QWidget` and then just disabling that widget plus the "Reset" button instead. The resulting git diff would be a bit annoying since everything would be indented by one more space, but it'd be much less and much cleaner code than this current PR.

**Comment by cg2121** [comment]
I've put the layout inside of a QFrame, so we can now just disable it instead.

**Comment by jp9000** [comment]
Why QFrame rather than QWidget?

**Comment by gxalpha** [comment]
To elaborate on Jim's comment, I don't see the value in using a QFrame here.

As stated by the [documentation](https://doc.qt.io/qt-6/qframe.html), "QFrame [...] is the base class of widgets that can have a frame". Using a QFrame with `NoFrame` is a bit pointless, it could just be a simple QWidget instead.
As far as I understand (and I'm by no means an expert), `NoFrame` is made for cases where you sometimes want a frame and sometimes not, and have to dynamically switch between them.

Of course, if there are specific QSS-related concerns that can only be solved with QFrames, it would be alright to use it; but if it's just because we use QFrame as a container everywhere that seems like a general problem across the program to me.

**Comment by cg2121** [comment]
QWidgets can't be styled with QSS.

**Comment by gxalpha** [comment]
> QWidget containers can't be styled with QSS.

I get that part.
What I'm asking is what the advantages of a QFrame are *in this specific scenario*, since - unless I'm mistaken - we're not looking to do any styling that would require a QFrame here. Is there something that requires the QFrame?

**Comment by cg2121** [comment]
Updated to change the QFrame to a QWidget.

File: UI/forms/OBSBasicTransform.ui [hunk 1]
```diff
@@ -6,213 +6,172 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>564</width>
-    <height>313</height>
+    <width>619</width>
+    <height>329</height>
```
Comment (gxalpha) [inline UI/forms/OBSBasicTransform.ui:10]: Is this change intentional?

File: UI/forms/OBSBasicTransform.ui [hunk 2]
```diff
@@ -226,592 +185,638 @@
...
-     <property name="orientation">
-      <enum>Qt::Vertical</enum>
-     </property>
-     <property name="sizeHint" stdset="0">
-      <size>
-       <width>0</width>
-       <height>10</height>
-      </size>
-     </property>
-    </spacer>
```
Comment (gxalpha) [inline UI/forms/OBSBasicTransform.ui:814]: Is the removal of this intentional?

**Review by sebastian-s-beckmann — COMMENTED** [review]
In addition to the two instances I marked below, there are a few buddy properties (cropLeft/cropBottom/cropTop/cropRight) getting removed. Are those intentional or just something Qt Creator decided to do on its own?
If they are intentional, could you quickly explain the motivation behind them? It doesn't *look* like anything's broken at first glance (We're not using ampersands for the label texts so the buddies currently [are likely a no-op anyways](https://doc.qt.io/qt-6/qlabel.html#setBuddy)) but a short explanation might be useful for future reference.

Other than that, this looks fine.

File: UI/forms/OBSBasicTransform.ui (hunk 1 above)
Comment (cg2121) [reply UI/forms/OBSBasicTransform.ui]: Fixed

File: UI/forms/OBSBasicTransform.ui (hunk 2 above)
Comment (cg2121) [reply UI/forms/OBSBasicTransform.ui]: Fixed

File: UI/forms/OBSBasicTransform.ui [hunk 3]
```diff
@@ -15,204 +15,157 @@
...
-        <property name="rightMargin">
-         <number>0</number>
-        </property>
-        <property name="bottomMargin">
-         <number>0</number>
-        </property>
-        <item>
-         <widget class="QDoubleSpinBox" name="positionX">
+    <widget class="QWidget" name="container" native="true">
+     <layout class="QGridLayout" name="gridLayout_2">
```
Comment (gxalpha) [inline UI/forms/OBSBasicTransform.ui:19]: Apologies for only seeing this now, but is this additional grid layout intentional?
I hope "Hide Whitespaces" isn't deceiving me here, but it looks like only (0, 0) of the layout gets filled with an item (the form layout), in which case I think the grid layout would be redundant? Hope I'm not missing something in the diff here.

Comment (cg2121) [reply UI/forms/OBSBasicTransform.ui:19]: A widget needs to have a layout to have other widgets inside of it.

Comment (gxalpha) [reply UI/forms/OBSBasicTransform.ui:19]: Sure, but - as far as I understand - here we have

`container` (widget)
\- `gridLayout_2` (layout)
\- \- [0, 0] `formLayout` (layout)

So gridLayout seems redundant, doesn't it?

Comment (cg2121) [reply UI/forms/OBSBasicTransform.ui]: Updated to remove the extra layout.

**Review by sebastian-s-beckmann — APPROVED** [review]
